File: python/Solution_Sword_11.py
class Solution:
    def minArray(self, numbers):
        l, r = 0, len(numbers) - 1
        while l < r:
            mid = ((r - l) >> 1) + l
            if numbers[r] > numbers[mid]:
                r = mid
            elif numbers[r] < numbers[mid]:
                l = mid + 1
            else:
                r -= 1
        return numbers[l]

if __name__ == '__main__':
    print(Solution().minArray([3,4,5,1,2]))

File: java/src/Solution_Sword_11.java
public class Solution_Sword_11 {
    public int minArray(int[] numbers) {
        int l = 0, r = numbers.length - 1;
        while (l < r) {
            int mid = ((r - l) >> 1) + l;
            if (numbers[r] > numbers[mid]) {
                r = mid;
            } else if (numbers[r] < numbers[mid]) {
                l = mid + 1;
            } else {
                --r;
            }
        }
        return numbers[l];
    }
}
